lib/listen/logger.rb

```diff
@@ -1,34 +1,35 @@
 # frozen_string_literal: true
 
 module Listen
-  def self.logger
-    @logger ||= nil
-  end
+  @logger = nil
 
-  def self.logger=(logger)
-    @logger = logger
-  end
+  # Listen.logger will always be present.
+  # If you don't want logging, set Listen.logger = ::Logger.new('/dev/null', level: ::Logger::UNKNOWN)
+
+  class << self
+    attr_writer :logger
 
-  def self.setup_default_logger_if_unset
-    self.logger ||= ::Logger.new(STDERR).tap do |logger|
-      debugging = ENV['LISTEN_GEM_DEBUGGING']
-      logger.level =
-        case debugging.to_s
-        when /2/
-          ::Logger::DEBUG
-        when /true|yes|1/i
-          ::Logger::INFO
-        else
-          ::Logger::ERROR
-        end
+    def logger
+      @logger ||= default_logger
```

Comment:
Maybe not thread safe :p

Reply:
Yes, I suppose 2 could wind up getting created, but I think we'd be ok, since they're equivalent.